pkg/storage/kubernetes.go

```diff
@@ -142,6 +142,65 @@ func (i *KubernetesIPAM) getPool(ctx context.Context, name string, iprange strin
 	return pool, nil
 }
 
+// IsAllocatedClusterWide checks for IP addresses to see if they're allocated cluster wide, for overlapping ranges
+func (i *KubernetesIPAM) IsAllocatedClusterWide(ctx context.Context, ip net.IP) (bool, error) {
+
+	// IPv6 doesn't make for valid CR names, so normalize it.
+	normalizedip := strings.ReplaceAll(fmt.Sprint(ip), "::", "-")
+
+	logging.Debugf("Clusterwide allocation check for IP: %v / namespace: %v", normalizedip, i.namespace)
+
+	clusteripres := &whereaboutsv1alpha1.ClusterIPReservation{
+		ObjectMeta: metav1.ObjectMeta{Name: normalizedip, Namespace: i.namespace},
```

Comment:
If this is a cluster-scoped resource, you shouldn't be setting the namespace

Reply:
Hey Chris, thanks for the eyes and definitely for the review, appreciate it a lot!

I went and played with it a `scope: Cluster` resource, and... I remembered that I had something in mind for a future enhancement which is this idea of "reuse of IPPools ranges across namespaces", I just put together some thoughts in #50 

So in this context... two parts:

1. I think when I go implement in openshift land, I'm going to have an easier time with having the CR namespaced.
2. I don't want to necessarily cut off design considerations because of that, but, I'm thinking in tandem with the considerations from #50 that I could also make these `ClusterIPReservations` namespaced as well, to essentially only mitigate the overlap when inside the same namespace. I would make this optional / configurable, too.

That being said... Now I think the name `ClusterIPReservation` is... Not great. Do you have any suggestions?

My first one is to make it a bit more specific and to not overload the term "cluster" and maybe something like...

`OverlappingIPRangeReservation`
`OverlappingRangeIPReservation`

or... Happy to take any input there, too.

Reply:
I tried a rename, I think it might express the idea better than how I overloaded the use of the word "cluster" here, really.